[INTER-1154] Proxy integration headers handling when proxy secret is missing

erayaydin opened the pull request:
Updated `addProxyIntegrationHeaders` to set proxy headers even if the proxy secret is not set. Adjusted test cases for new behavior.

Related-Task: INTER-1154

erayaydin committed:
feat: proxy integration headers handling when proxy secret is missing

Updated `addProxyIntegrationHeaders` to set proxy headers even
proxy secret is not set. Adjusted test cases for new behavior.

Related-Task: INTER-1154

JuroUhlar commented on test/utils/addProxyIntegrationHeaders.test.ts:
```diff
@@ -33,8 +33,8 @@ describe('addProxyIntegrationHeaders', () => {
     expect(headers.get('FPJS-Proxy-Secret')).toBe(null)
-    expect(headers.get('FPJS-Proxy-Client-IP')).toBe(null)
-    expect(headers.get('FPJS-Proxy-Forwarded-Host')).toBe(null)
+    expect(headers.get('FPJS-Proxy-Client-IP')).toBe(ipv4)
```
You should update the test description on line 32 as well

erayaydin replied:
I didn't notice, nice catch!

erayaydin replied:
Updated test name

JuroUhlar commented on test/endpoints/getResult.test.ts:
```diff
@@ -310,29 +310,33 @@ describe('ingress API request headers', () => {
   })
 
-  test('req headers are the same (except Cookie) when no proxy secret', async () => {
+  test('req headers are the same (except proxy headers) when no proxy secret', async () => {
```
Clarity suggestion

```suggestion
  test('Fpjs-Proxy-Client-Ip and Fpjs-Proxy-Forwarded-Host headers are added to the proxied request headers even if proxy secret is undefined ', async () => {
```

erayaydin replied:
@JuroUhlar I want to prevent the creation of new tests for the potential new proxy headers. Mentioning only **Client-IP** and **Forwarded-Host** might push the developer to create a new test instead of editing this one. I could be wrong, but do you have any idea about this? To make it clear, should we use `FPJS-Proxy-Client-IP and FPJS-Proxy-Forwarded-Host headers are added ...` or `FPJS-Proxy headers are added ...` / `Proxy headers are added ...`? We can also check this in the reviews whether the developers have updated this test. Both options seem fine to me. We can proceed based on your guidance 😇 

> Additionally, this test checks whether the custom headers are preserved. We'll need to mention this in the test name as well

JuroUhlar replied:
Sure,  you can say something like
```suggestion
  test('Even if proxy secret is undefined, other Fpjs-Proxy-* headers are still added to the proxied request headers. Original headers are preserved.', async () => {
```

JuroUhlar reviewed (changes requested):
2 suggestions

erayaydin committed:
test: update test names for FPJS-Proxy-* headers behavior

Renamed test cases to clarify that FPJS-Proxy-* headers are added
even when the proxy secret is `undefined` or `null`. Also ensured
the test names explicitly state that original headers are preserved

Co-Authored-By: @JuroUhlar
Related-Task: INTER-1154
